src/daemon.js

```diff
@@ -113,6 +115,9 @@ class Daemon {
         options.dht && execOptions.push('--dht')
         options.pubsub && execOptions.push('--pubsub')
       }
+      if ((this._keyFile || '') !== '') {
```

Comment:
I would just pass the key in the start call and do `if (options.id) { /* push key to exec */. }`.

@vasco-santos I think we should just do reverse arg parsing here, with something like https://github.com/sindresorhus/dargs, so that we don't have to special case everything. I know go args use single `-`'s, but perhaps we could do some post processing on the args list to clean modify that based on daemon type. What do you think?

Reply:
I did not spend much time thinking about this parse since I wanted to move everything to a JSON config, once we support that. However, as it is an easy change to use `dargs` and post-process the resulting array if GO, we can modify this here